localstack/utils/collections.py

```diff
@@ -109,6 +112,70 @@ def __hash__(self):
         return hash(canonical_json(self._dict))
 
 
+class DictProxy(MutableMapping):
+    """
+    Generic dict proxy class that can be used to define subclasses of dict with additional methods.
+    It implements the MutableMapping interface and redirects all methods calls to an underlying dict which it wraps.
+
+    The DictProxy class can be extended with additional custom logic as follows:
```

Comment:
Afaik the regular built-in `dict` can already do this. Is there some extra reason for this apart from adding new methods?

Reply:
Indeed, the built-in airspeed dict already supports `put`, nice catch. This was more to anticipate other additional map methods like `putAll`, `entrySet`, etc. 

Just realized that it may be cleaner to patch `__additional_methods__` in airspeed directly for those cases. Will remove the DictProxy again for now, we can revert to it later once a concrete use case arises.